### ISSUE TITLE
Outgoing connection to Azure SQL Server are closed on Consumption work profile

### ISSUE THREAD

**Issue by claria** [microsoft/azure-container-apps#1338]
### This issue is a: (mark with an x)
- [ ] bug report -> please search issues before submitting
- [ ] documentation issue or request
- [x] regression (a behavior that used to work and stopped in a new release)

### Issue description

We are running an azure container app on the consumption work profile. (1vcpu/2GB Ram) and is running with 2 instances. The deployed docker container is a Keycloak v26 release. Keycloak connects to our Azure SQL database. The container keeps open a connection pool of 20 connections to the sql database per replica. 

Since around 2-3 weeks, we suddenly see "Connection reset by peer" errors when these connections are tried to beused. If i understand correctly, something inbetween (or the Azure sql server itself) closed the connections. 

Since we initially thought, that this was due to killing of idling connections, we enabled sql connection validation and configured it that every 10 seconds a "select 1" statement is sent for each connection to keep it alive and validate it. We see on the sql server side that it works and each connection is used every 10 seconds.
However, the problem continues and we see the same amount of connection reset by peer errors. 

Our current investigations are:

- On Azure SQL side we are far away from any connection limits etc.
- All zure applications connection to this Azure sql server behave normal.
- This behaviour is not happending all the time. Sometimes, we do not see any dropped connections for a couple of hours, then it starts again.
- We deploy the exact same docker container/configuration from an azure VM. The problem does not occur. Everything is stable.
- We deploy the exact same configuration but with a dedicated work profile. The problem does not occur. Everything is stable

### Steps to reproduce
1. Create Container app with consupmtion work profile
2. Deploy Keycloak container

**Expected behavior** [What you expected to happen.]

- Connections to sql server are not closed.

**Actual behavior** [What actually happened.]

The connection to the sql server remains open (if it is active)

**Screenshots**         

### Additional context
Ex. Did this issue occur in the CLI or the Portal?

**Comment by JennyLawrance** [comment]
Hi @claria , can you send us the ARM resource URI to acasupport(at)microsoft(dot)com? We will investigate.

**Comment by claria** [comment]
I sent the resource uri via mail.

**Comment by JennyLawrance** [comment]
[like]  Jenny Lawrance reacted to your message:
________________________________

**Comment by claria** [comment]
This is a count of all the "connection reset by peer" errors we see. We have two replica of the container app running. As you can see sometimes the error do not occure for over an hour for one replica while still happening for the other:

Logs are written to a log analytics workspace. Please have a look by yourself if you want.

![Image](https://github.com/user-attachments/assets/af14874e-a649-4b3d-8adb-b4a2df331c93)

**Comment by claria** [comment]
Any update @JennyLawrance ?

**Comment by JennyLawrance** [comment]
Yes, We applied a fix on your environment last week. Let us know if you see improvement in the connection status.

**Comment by claria** [comment]
Hi @JennyLawrance,

no, unfortunately the situation did not improve. These are the experienced connection error counts over the last week which have not really changed:
![Image](https://github.com/user-attachments/assets/958e268c-d744-4e5d-aa49-ca355a01671b)

**Comment by JennyLawrance** [comment]
Hi @claria , I was able to confirm that we had an internal crossing of wires, and the fix wasn't applied to this environment as I previously stated. I apologize for the miscommunication here. We applied it today again (and I confirmed that at my side.) 
Can you check the status on your end and let us know what you see in your logs?

Thanks, Jenny

**Comment by claria** [comment]
Hi @JennyLawrance,

i can confirm that the errors are completely gone after your fix. What did you change and how can we prevent the errors to happen in the future?

We plan to create multiple similar environments. Do we have to expect the same error to happen there?

**Comment by claria** [comment]
@JennyLawrance Could you please confirm if this fix was applied globally or only on this one affected instance?

**Comment by JennyLawrance** [comment]
@claria Thanks for your patience. We have applied the fix to West Europe, all newly created Container App replicas will get the fix.

@jsheetzmt , can you send across your Azure resource ID for the impacted instance? We will investigate and find out if you are running into the same root cause as the one discussed for West Europe.
